index.d.ts

```diff
@@ -10,226 +10,233 @@ declare global {
+
+  /**
+   * Connect to ArConnect and request permissions. This function can always be
+   * called again if you want to request more permissions for your site.
+   *
+   * @param permissions
+   * @param appInfo
+   */
+  connect(
+    permissions: PermissionType[],
+    appInfo?: AppInfo,
+    gateway?: GatewayConfig
+  ): Promise<void>;
+
+  /**
+   * Disconnect from ArConnect. Removes all permissions from your site.
+   */
+  disconnect(): Promise<void>;
+
+  /**
+   * Get the currently used wallet's address in the extension.
+   *
+   * @returns Promise of wallet address string
+   */
+  getActiveAddress(): Promise<string>;
+
+  /**
+   * Get all addresses added to the ArConnect extension
+   *
+   * @returns Promise of a list of the added wallets' addresses.
+   */
+  getAllAddresses(): Promise<string[]>;
+
+  /**
+   * Get wallet names for addresses.
+   *
+   * @returns Promise of an object with addresses and wallet names
+   */
+  getWalletNames(): Promise<{ [addr: string]: string }>;
+
+  /**
+   * Sign a transaction.
+   *
+   * @param transaction A valid Arweave transaction without a wallet keyfile added to it
+   * @param options Arweave signing options
+   *
+   * @returns Promise of a signed transaction instance
+   */
+  sign(
+    transaction: Transaction,
+    options?: SignatureOptions
+  ): Promise<Transaction>;
+
+  /**
+   * Get the permissions allowed for you site by the user.
+   *
+   * @returns Promise of a list of permissions allowed for your dApp.
+   */
+  getPermissions(): Promise<PermissionType[]>;
+
+  /**
+   * Encrypt a string, using the user's wallet.
+   *
+   * @param data Buffer-like structure to encrypt.
+   * @param options Encrypt options. See https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt#algorithm.
+   *
+   * @returns Promise of the encrypted `Uint8Array`.
+   */
+  encrypt(
+    data: BinaryDataType,
+    options?: RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams,
+  ): Promise<Uint8Array>;
+
+  /**
+   * Decrypt a string encrypted with the user's wallet.
+   *
+   * @param data Buffer-like structure to decrypt.
+   * @param options Decrypt options. See https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/decrypt#algorithm.
+   *
+   * @returns Promise of the decrypted `Uint8Array`.
+   */
+  decrypt(
+    data: BinaryDataType,
+    options?: RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams,
+  ): Promise<Uint8Array>;
+
+  /**
+   * Get the user's custom Arweave config set in the extension
+   *
+   * @returns Promise of the user's Arweave config
+   */
+  getArweaveConfig(): Promise<GatewayConfig>;
+
+  /**
+   * @deprecated Find alternatives at https://docs.arconnect.io/api/signature
+   */
+  signature(
+    data: BinaryDataType,
+    // https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign#algorithm
+    algorithm?: RsaPssParams | EcdsaParams | { name: "RSASSA-PKCS1-v1_5" } | { name: "HMAC" },
+  ): Promise<Uint8Array>;
```

Comment:
Before:

```
      signature(
        data: Uint8Array,
        // https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign#parameters
        algorithm: AlgorithmIdentifier | RsaPssParams | EcdsaParams
      ): Promise<Uint8Array>;
```

But `AlgorithmIdentifier` is more broad than what's defined in the docs.